ocaml/xapi/xapi_ha.ml

```diff
@@ -1668,6 +1668,7 @@ let disable __context =
   let pool = Helpers.get_pool ~__context in
   if not (Db.Pool.get_ha_enabled ~__context ~self:pool) then
     raise (Api_errors.Server_error (Api_errors.ha_not_enabled, [])) ;
+  Xapi_pool_helpers.assert_designate_new_master_not_in_progress ~__context ;
```

Comment:
I think this is redundant, you should've been blocked from starting the designate operation if HA is enabled. Also HA does designate new masters on its own.